### ISSUE TITLE
Not working on stable 88.0.1

### ISSUE THREAD

**Issue by Polisson** [thepante/SAL-Firefox#2]
Hi, 
Firstly thanks for your work.

Few things are mixed up here.
I'm using:
Name 	| Firefox
------------ | ------------- 
Version 	| 88.0.1
Build ID 	| 20210504152106
Update Folder 	| C:\ProgramData\Mozilla\updates\CB615D7E7242D42C
Update Channel 	| release
User Agent 	| Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:88.0) Gecko/20100101 Firefox/88.0
OS 	| Windows_NT 10.0 19042
Application Binary 	| P:\_internet\_browsers\FirefoxPortable\App\firefox64\firefox.exe

1. Your wiki doesn't mention right placing files for 64bit arch as some Windows users might need,
2. the script loader provided by You doesn't work (probably the line of `config-prefs.js` :
 `pref("general.config.sandbox_enabled", false);`) 
 prevents from loading scripts. 
When I changed do _alice0775's repo_ the  FloatingScrollbar.uc.js script started to loading fine,
3. Your `salf.uc.js` seems to be loaded because when I set enable to shortcut `Ctrl+e`, the panel is being hidden/unhidden,
4. Panel show up and hides, with enabling by button, but always is reloaded, so general functionality of your script is absent

I would provide further info if needed.
Greetings
Mat

**Comment by thepante** [comment]
Hi there Mat. 
Sadly is working on 89 and up only. Seems that Mozilla made changes in the Firefox interface design that make a break between that version and the 88.

In fact, the changes that they made on the 89 seems to be the reason why I finally could re-do the float mode to end up with the script described in the readme as today. Before this, I tried to do it in 88 and some previous versions without luck.

For 88 should work the basic [2.5 version of this script](https://raw.githubusercontent.com/thepante/SAL-Firefox/2.5/salf.uc.js), but doesn't have shortcut and it is "classic" mode only (doesn't have the option to float). 

On the other side, `FloatingScrollbar.uc.js` thankfully didn't get affected as this one between versions changes. So thats why runs properly on *recent* versions (oldest one that I have record is 63.0a1, currently on 90.0a1). Had to adjust some details but they didn't make breaking changes about scrollbars.

Firefox 89 release got delayed a little bit and now is set to [release on June 1 2021](https://wiki.mozilla.org/Release_Management/Release_owners). You could try the 2.5 basic script version or wait and try out the most recent one with the 89 when you got the update.

---

About the loader, the version I use and the one zipped in the wiki is the 3.0.59 which doesn't use the line `pref("general.config.sandbox_enabled", false);` in `config-prefs.js`. That line seems to be for 3.0.60 version in alice0775's repo.

All the files of the loader should be from the same version: 3.0.59 (in .zip) or 3.0.60 (latest commit in author's repo), but not mixed.

---

About the path:
> 1. Your wiki doesn't mention right placing files for 64bit arch as some Windows users might need,

Can I know which path did you ended up using? With that info it should be `P:_internet_browsers\FirefoxPortable\App\firefox64\` - wasn't that?

**Comment by Polisson** [comment]
Hi, sorry for so much lag in response.
Firstly, I ended with `firefox64` path. Copying files to regular `FirefoxPortable\App\Firefox` seems not to work at all. Also, I hadn't mixed yours and alice0775's repos.
But...!
I've updated FF to recent stable version which is 89.0. Still the `FloatingScrollbar.uc.js` functionality is broken with your _user script loader_
Unfortunately, `salf.uc.js` doesn't work for me, no matter whose loader I try and no matter if I start my Firefox as portable or as standalone (with clean profile in _user roaming path_).

How can I help you to investigate this?

**Comment by thepante** [comment]
Hi there. The `about:support` page should list the path of the firefox binary in (actual) use, which for what I undestand; it is pointing to the correct one (firefox64). 

Anyway, I would update the wiki to avoid confusion about *paths thing*. Thanks for bring that up.

---

I don't know what happen, but now I tried firefox 89 with the loader I actually use and doesn't work. Tried with the newest (from alice0775's repo) and the loader works, but not this script.

I think that I found the issue. But the error I'm seeing isn't exactly the one you described in the first message. Would try to fix that and let you know when it is ready to download to try it out. 

Alongside with the (I hope) fixed version, I would update the .zip with the newest loader.

**Comment by thepante** [comment]
Done, at least I got it fixed on my firefox 89 fresh install & profile.
Try it out ([salf-userchrome.zip](https://github.com/thepante/SAL-Firefox/releases/download/3.3/salf-userchrome.zip)) and let me know the results.

Tested with the loader v3.0.60, updated on that wiki. Also attached on [3.3 release](https://github.com/thepante/SAL-Firefox/releases/tag/3.3)

**Comment by Polisson** [comment]
Yes! Works after update (first time for me, so I'm very excited 8)).
Great option of automatic closing panel. Wondering how to make it even more usable (e.g. partially visible and hover to show it)
Thanks!

**Comment by thepante** [comment]
Great! Thank you for helping out with the issue.

> (e.g. partially visible and hover to show it)

That sounds great. I think shouldn't be problems to add this. I'll try to implement this on these days and let you know!

**Comment by thepante** [comment]
Done with that option @Polisson. Check it out on [3.4](https://github.com/thepante/SAL-Firefox/releases/tag/3.4) and let me know if there's something to adjust.
Enable it changing to `true` the `part_hide` → `enabled` key.
